engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnStateTransitionBehavior.java

```diff
@@ -266,6 +287,26 @@ private boolean canActivateTargetElement(
           elementInstanceState.getNumberOfTakenSequenceFlows(
               context.getFlowScopeKey(), targetElement.getId());
       return numberOfTakenSequenceFlows == numberOfIncomingSequenceFlows;
+    } else if (targetElement.getElementType() == BpmnElementType.INCLUSIVE_GATEWAY) {
+      // activate the inclusive gateway  if some  incoming sequence flows are taken
+      if (numberOfIncomingSequenceFlows == 1) {
+        return true;
+      } else {
+        final int numberOfTakenSequenceFlows =
+            elementInstanceState.getNumberOfTakenSequenceFlows(
+                context.getFlowScopeKey(), targetElement.getId());
+
+        final int numberOfActiveElementInstances =
+            elementInstanceState
+                .getInstance(context.getFlowScopeKey())
+                .getNumberOfActiveElementInstances();
+
+        if (context.getBpmnElementType() == BpmnElementType.INCLUSIVE_GATEWAY) {
+          return numberOfTakenSequenceFlows == numberOfIncomingSequenceFlows;
+        } else {
+          return numberOfActiveElementInstances == 0;
+        }
+      }
```

Comment:
👍 I think this will work, as long as we manage to correctly keep track of the number of taken sequence flows.

I believe that these 2 problems lay at the core of making the inclusive gateway work, and why we always considered it hard to implement. I think you've just solved it.

👏 Well done!

Reply:
`return numberOfActiveElementInstances == 0;`

❌ I've discovered that this isn't gonna be enough.

Consider the following process. Let's say we don't complete the job related to the service task. We'd expect the joining inclusive gateway to activate after the manual tasks have completed. However, it doesn't because the service task is an active flow in the flowscope of the gateway (i.e. in this case the flowscope is the process instance itself). So the `numberOfActiveElementInstances` is `1` and not `0`.
<img width="785" alt="Screen Shot 2022-07-15 at 18 25 33" src="https://user-images.githubusercontent.com/3511026/179266156-419934ff-9a27-4ea0-b2c6-82eae8b91aab.png">

The problem is that the inclusive gateway should only be activated if tokens exist that can still reach the gateway. This token exists in the flow scope, but it can't reach the gateway. We need to actually look for directed paths between any of the children of the flowscope (you can use `ElementInstanceState.getChildren(key)`) and the gateway.

Reply:
I add a test case for this `InclusiveGatewayTest.shouldMergeInclusiveBranchesWithParallelGateway` .